### DISCUSSION TITLE
Being able to use Astro code fence (---) on other frameworks

### DISCUSSION THREAD

**Discussion by katriellucas** [withastro/roadmap#255] [Proposal]
This is just an idea I had, not sure how feasible it's.
I believe there is currently no way to achieve the hybrid approach of Astro components (fenced code + script tag) without using a second component on client:load for other frameworks like Svelte, if I need reactivity on the component and also query a database, I think I have to do something like this:
```js
<SvelteOuter>
  <SvelteInner client:load/>
</SvelteOuter>
```
On other hand, Astro components just need to fence the database query, and do whatever they need on a script tag (tho no reactivity), you can make everything in a single component.

In the end, with other frameworks, it's either build rendered or client rendered, where Astro components you can mix both, would not it be nice to be able to do this with other frameworks as well?

**Comment by L1lith** [comment]
This would be pretty awesome!
